enhancements/network/requesting-staticips-for-vms-being-migrated-into-primary-l2-udns-using-MTV.md

```diff
@@ -0,0 +1,849 @@
+the claim has been consumed by a different pod. Same for VM start/stop
+scenarios.
+
+Below you'll see the changes requested:
+```go
+
+// IPAMClaimStatus contains the observed status of the IPAMClaim.
+type IPAMClaimStatus struct {
+    // The list of IP addresses (v4, v6) that were allocated for the pod interface
+    IPs []string `json:"ips"`
++   // The name of the pod holding the IPAMClaim
++   OwnerPod string `json:"ownerPod"`
++   // Conditions contains details for one aspect of the current state of this API Resource
++   Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+The `IPAMClaim` status will have (at least) the following conditions:
+- IPsAllocated: when the status is "True", it reports the IP address was
+  successfully allocated for the workload. The reason will be
+  `SuccessfulAllocation`. When the status is "False", it reports an IP
+  allocation error, with a specific `reason`, and more details will be provided
+  in the `message` attribute.
+
+The reason enums are described below:
+- IPAddressConflict
+- RequestedIPOutsideSubnet
+- ReservedIPRequested (in case we want to be pedantic and differentiate with
+  the IPs reserved for management port, gateway, or the subnet excluded IPs)
+
+These conditions will look like - **successful** allocation:
+```
+type: IPsAllocated
+status: "True"
+reason: SuccessfulAllocation
+message: "IP 192.168.1.5 allocated successfully"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+These conditions will look like - **failed** allocation:
+```
+type: IPsAllocated
+status: "False"
+reason: IPAddressConflict
+message: "Requested IP 192.168.1.5 is already assigned in the network"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+```
+type: IPsAllocated
+status: "False"
+reason: RequestedIPOutsideSubnet
+message: "Requested IP 200.168.1.5 is outside the UDN subnet 192.168.0.0/16"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+```
+type: IPsAllocated
+status: "False"
+reason: ReservedIPRequested
+message: "The user requested IP address 192.168.200.123 which was explicitly excluded by the admin"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The management cluster should have no reason to import an existing VM from
+another virtualization platform into OpenShift Virtualization.
+
+It makes more sense to import VMs into the hosted clusters, in order to provide
+tenants with routed ingress into their hosted cluster workloads.
+
+There should be no hypershift platform-specific considerations with this
+feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+Full support; there might be some increased resource usage when scaling up
+number of VMs requesting IP and MAC address, since it will mean more processing
+in OVN-Kubernetes, and more CRs to reconcile.
+
+### Implementation Details/Notes/Constraints
+
+N/A
```

Comment:
I can't approve the enhancement with this part not filled... Implementation details on kubevirt/mtv has to be part of the enhancement right? if not I would appreciate at least a comment saying why its "not applicable"

Reply:
https://github.com/openshift/enhancements/pull/1793#discussion_r2204542948

Reply:
Ah I think you need to put - the implementation details are in the Proposal section so already covered.. TODO or NA has different meanings